src/java.base/share/classes/jdk/internal/classfile/impl/BytecodeHelpers.java

```diff
@@ -56,83 +57,125 @@ public static IllegalArgumentException cannotConvertException(TypeKind from, Typ
 
     public static Opcode loadOpcode(TypeKind tk, int slot) {
         return switch (tk) {
-            case INT, SHORT, BYTE, CHAR, BOOLEAN -> switch (slot) {
-                case 0 -> Opcode.ILOAD_0;
-                case 1 -> Opcode.ILOAD_1;
-                case 2 -> Opcode.ILOAD_2;
-                case 3 -> Opcode.ILOAD_3;
-                default -> (slot < 256) ? Opcode.ILOAD : Opcode.ILOAD_W;
-            };
-            case LONG -> switch (slot) {
-                case 0 -> Opcode.LLOAD_0;
-                case 1 -> Opcode.LLOAD_1;
-                case 2 -> Opcode.LLOAD_2;
-                case 3 -> Opcode.LLOAD_3;
-                default -> (slot < 256) ? Opcode.LLOAD : Opcode.LLOAD_W;
-            };
-            case DOUBLE -> switch (slot) {
-                case 0 -> Opcode.DLOAD_0;
-                case 1 -> Opcode.DLOAD_1;
-                case 2 -> Opcode.DLOAD_2;
-                case 3 -> Opcode.DLOAD_3;
-                default -> (slot < 256) ? Opcode.DLOAD : Opcode.DLOAD_W;
-            };
-            case FLOAT -> switch (slot) {
-                case 0 -> Opcode.FLOAD_0;
-                case 1 -> Opcode.FLOAD_1;
-                case 2 -> Opcode.FLOAD_2;
-                case 3 -> Opcode.FLOAD_3;
-                default -> (slot < 256) ? Opcode.FLOAD : Opcode.FLOAD_W;
-            };
-            case REFERENCE -> switch (slot) {
-                case 0 -> Opcode.ALOAD_0;
-                case 1 -> Opcode.ALOAD_1;
-                case 2 -> Opcode.ALOAD_2;
-                case 3 -> Opcode.ALOAD_3;
-                default -> (slot < 256) ? Opcode.ALOAD : Opcode.ALOAD_W;
-            };
-            case VOID -> throw new IllegalArgumentException("void");
+            case INT, SHORT, BYTE, CHAR, BOOLEAN
+                           -> iload(slot);
```

Comment:
Can you do
```java
switch (tk.asLoadable()) {
    case INT      -> iload(slot);
```

Reply:
Why can it be simplified? I don't understand

Reply:
Logically I think `asLoadable` is clearer and reduces switch size.
And enum `ordinal` field is already constant-folded: https://bugs.openjdk.org/browse/JDK-81612454
So it can be elided by JIT if the enum argument is constant.